### ISSUE TITLE
"Signal keeps stopping"

### ISSUE THREAD

**Issue by jperryhouts** [signalapp/Signal-Android#8180]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
Signal won't open. At all. It consistently gives me 'Signal keeps stopping' errors.

### Steps to reproduce
- Open Signal
- Become frustrated

**Actual result:**
App crashes. Android gives me a dialog that says Signal keeps stopping.

**Expected result:**
App works.

### Device info

**Device:** Google Pixel
**Android version:** 9 (Security patch level: August 5, 2018)
**Signal version:** 4.24.8

### Link to debug log
I can't get far enough to take a debug log. Here's snippets from the system log.

I think this is the important part:
```
09-06 12:25:18.389 10110  6839  6862 W JobConsumer: null
09-06 12:25:18.389 10110  6839  6862 W JobConsumer: java.lang.NullPointerException: Attempt to invoke virtual method 'java.io.FileInputStream android.content.res.AssetFileDescriptor.createInputStream()' on a null object reference
09-06 12:25:18.389 10110  6839  6862 W JobConsumer: 	at org.thoughtcrime.securesms.jobs.MultiDeviceContactUpdateJob.getAvatar(MultiDeviceContactUpdateJob.java:244)
09-06 12:25:18.389 10110  6839  6862 W JobConsumer: 	at org.thoughtcrime.securesms.jobs.MultiDeviceContactUpdateJob.generateSingleContactUpdate(MultiDeviceContactUpdateJob.java:119)
09-06 12:25:18.389 10110  6839  6862 W JobConsumer: 	at org.thoughtcrime.securesms.jobs.MultiDeviceContactUpdateJob.onRun(MultiDeviceContactUpdateJob.java:103)
09-06 12:25:18.389 10110  6839  6862 W JobConsumer: 	at org.thoughtcrime.securesms.jobs.MasterSecretJob.onRun(MasterSecretJob.java:18)
09-06 12:25:18.389 10110  6839  6862 W JobConsumer: 	at org.thoughtcrime.securesms.jobmanager.JobConsumer.runJob(JobConsumer.java:74)
09-06 12:25:18.389 10110  6839  6862 W JobConsumer: 	at org.thoughtcrime.securesms.jobmanager.JobConsumer.run(JobConsumer.java:47)
09-06 12:25:18.390 10110  6839  6862 E AndroidRuntime: FATAL EXCEPTION: JobConsumer-3
09-06 12:25:18.390 10110  6839  6862 E AndroidRuntime: Process: org.thoughtcrime.securesms, PID: 6839
...

09-06 12:25:18.393  1000   875  5561 W ActivityManager:   Force finishing activity org.thoughtcrime.securesms/.RoutingActivity
09-06 12:25:18.405  1000   875   914 I ActivityManager: Showing crash dialog for package org.thoughtcrime.securesms u0
```
And here are a few other related parts of the debug log:
```
  Time since processes crashed:
    Process org.thoughtcrime.securesms uid 10110: last crashed +21s814ms ago
```
```
Recent Sync History
(SERVER is now split up to FEED and OTHER)
  #1  : 2018-09-06 12:26:10    OTHER   13.5s            Signal/org.thoughtcrime.securesms u0                                              com.android.contacts                        ServiceChanged
  #2  : 2018-09-06 12:24:50    OTHER    7.3s     01:19  Signal/org.thoughtcrime.securesms u0                                              com.android.contacts                        ServiceChanged
    mesg=io-error
```
```
** Database info for pid 7775 [org.thoughtcrime.securesms] **

Connection pool for /data/user/0/org.thoughtcrime.securesms/databases/_jobqueue-TextSecureJobs:
  Open: true
  Max connections: 1
  Total execution time: 10
  Configuration: openFlags=268435456, useCompatibilityWal=true, journalMode=, syncMode=
  Available primary connection:
    Connection #0:
      isPrimaryConnection: true
      onlyAllowReadOnlyOperations: false
      Most recently executed operations:
        0: [2018-09-06 12:26:23.616] executeForLastInsertedRowId took 0ms - succeeded, sql="INSERT INTO queue(item,encrypted) VALUES (?,?)"
        1: [2018-09-06 12:26:23.616] prepare took 1ms - succeeded, sql="INSERT INTO queue(item,encrypted) VALUES (?,?)"
        2: [2018-09-06 12:26:14.719] executeForCursorWindow took 0ms - succeeded, sql="PRAGMA database_list;"
        3: [2018-09-06 12:26:14.715] executeForLong took 0ms - succeeded, sql="PRAGMA page_size;"
        4: [2018-09-06 12:26:14.715] executeForLong took 0ms - succeeded, sql="PRAGMA page_count;"
        5: [2018-09-06 12:26:10.315] executeForCursorWindow took 0ms - succeeded, sql="SELECT * FROM queue WHERE encrypted = 0 ORDER BY _id ASC"
        6: [2018-09-06 12:26:10.315] prepare took 0ms - succeeded, sql="SELECT * FROM queue WHERE encrypted = 0 ORDER BY _id ASC"
        7: [2018-09-06 12:26:10.315] executeForLong took 0ms - succeeded, sql="PRAGMA user_version;"
        8: [2018-09-06 12:26:10.314] prepare took 0ms - succeeded, sql="PRAGMA user_version;"
        9: [2018-09-06 12:26:10.313] executeForString took 1ms - succeeded, sql="SELECT locale FROM android_metadata UNION SELECT NULL ORDER BY locale DESC LIMIT 1"
        10: [2018-09-06 12:26:10.313] execute took 0ms - succeeded, sql="CREATE TABLE IF NOT EXISTS android_metadata (locale TEXT)"
        11: [2018-09-06 12:26:10.313] executeForLong took 0ms - succeeded, sql="PRAGMA wal_autocheckpoint=100"
        12: [2018-09-06 12:26:10.313] executeForLong took 1ms - succeeded, sql="PRAGMA wal_autocheckpoint"
        13: [2018-09-06 12:26:10.313] executeForLong took 0ms - succeeded, sql="PRAGMA journal_size_limit=524288"
        14: [2018-09-06 12:26:10.312] executeForLong took 0ms - succeeded, sql="PRAGMA journal_size_limit"
        15: [2018-09-06 12:26:10.309] execute took 3ms - succeeded, sql="PRAGMA synchronous=NORMAL"
        16: [2018-09-06 12:26:10.309] executeForString took 0ms - succeeded, sql="PRAGMA synchronous"
        17: [2018-09-06 12:26:10.305] executeForString took 4ms - succeeded, sql="PRAGMA journal_mode"
        18: [2018-09-06 12:26:10.305] executeForLong took 0ms - succeeded, sql="PRAGMA foreign_keys"
        19: [2018-09-06 12:26:10.305] executeForLong took 0ms - succeeded, sql="PRAGMA page_size"
  Available non-primary connections:
    <none>
  Acquired connections:
    <none>
  Connection waiters:
    <none>
```
```
    Process: org.thoughtcrime.securesms/PID: 5650/Flags: 0x38d83e44/Packag ...
2018-09-05 10:48:54 keymaster (data, 53 bytes)
2018-09-05 10:48:54 keymaster (data, 53 bytes)
2018-09-05 10:48:54 keymaster (data, 53 bytes)
2018-09-05 10:48:55 data_app_crash (text, 979 bytes)
    Process: org.thoughtcrime.securesms/PID: 1538/Flags: 0x38d83e44/Packag ...
```

**Comment by greyson-signal** [comment]
Fixed in 4.26

**Comment by jperryhouts** [comment]
Thank you!
